kivy/graphics/vertex_instructions.pyx

```diff
@@ -1592,3 +1626,728 @@ cdef class RoundedRectangle(Rectangle):
     def radius(self, value):
         self._radius = self._check_radius(value)
         self.flag_data_update()
+
+
+
+
+"""
+Graphics section with antialiasing that uses a combination of AntiAliasingLine
+and the target graphics Instruction, such as Rectangle, Ellipse,
+RoundedRectangle, Triangle and Quad.
+
+NOTE: Texture antialiasing is currently not supported. If a texture is defined
+for any of the graphics with antialiasing, then antialiasing will be disabled.
+
+The antialiasing is also disabled for graphics with "fixed" shapes, such as
+Rectangle, RoundedRectangle and Ellipse if -4 < width < 4px
+or -4 < height < 4px. Reasons for it:
+
+    - Drawing an antialiasing line on figures with very small dimensions does
+    not bring great visual improvements. 
+
+    - This reduces the code complexity in the `adjust_params` functions,
+    which are used to adjust the size of these figures, and keep them
+    proportional to the figures without antialiasing.
+
+TODO: Use AntiAliasingLine as a sort of "alpha test" to enable texture
+antialiasing. This will likely involve utilizing glBlendFunc in conjunction
+with other functions. It would also involve creating custom instructions,
+similar to the custom stencil instructions bellow, to ensure efficiency.
+"""
+
+
+
+
+"""
+Custom stencil instructions used in AntiAliasingLine.
+It is basically identical to the instructions in the stencil_instructions.pyx
+module, however, there is a limitation on the number of calls to the functions
+that clear the stencil. This significantly improved performance with no
+noticeable side effects!
+"""
+
+cdef class AAStencilPush(Instruction):
+    cdef int apply(self) except -1:
+        cgl.glStencilMask(0xff)
+        if self.flags & GI_NEEDS_UPDATE:
+            cgl.glClearStencil(0)
+            cgl.glClear(GL_STENCIL_BUFFER_BIT)  # GL_STENCIL_BUFFER_BIT is a bit expensive, so its use will be limited.
+        cgl.glEnable(GL_STENCIL_TEST)
+        cgl.glStencilFunc(GL_ALWAYS, 1, 0xff)
+        cgl.glStencilOp(GL_INCR, GL_INCR, GL_INCR)
+        cgl.glColorMask(False, False, False, False)
+        return 0
+
+
+cdef class AAStencilUse(Instruction):
+    cdef int apply(self) except -1:
+        cgl.glColorMask(True, True, True, True)
+        cgl.glStencilFunc(GL_GREATER, 1, 0xff)
+        cgl.glStencilOp(GL_KEEP, GL_KEEP, GL_KEEP)
+        return 0
+
+
+cdef class AAStencilUnUse(Instruction):
+    cdef int apply(self) except -1:
+        cgl.glStencilFunc(GL_GREATER, 0xff, 0xff)
+        cgl.glStencilOp(GL_DECR, GL_DECR, GL_DECR)
+        cgl.glColorMask(False, False, False, False)
+        return 0
+
+
+cdef class AAStencilPop(Instruction):
+    cdef int apply(self) except -1:
+        cgl.glColorMask(True, True, True, True)
+        cgl.glDisable(GL_STENCIL_TEST)
+        cgl.glStencilFunc(GL_EQUAL, 1, 0xff)
+        cgl.glStencilOp(GL_KEEP, GL_KEEP, GL_KEEP)
+        return 0
```

Comment:
Since most of the code is duplicated from stencil instructions, and I agree that introducing the check to avoid calling `cgl.glClear(GL_STENCIL_BUFFER_BIT)` every time makes sense also on standard `StencilPush`, how about introducing the feature in `StencilPush`, but with a setting that enables it? (disabled by default)

That is actually great to conform to DRY principle, and developers and maintainers can take advantage of the speedup if they want to.

Reply:
I think it's a good idea!